### PR TITLE
Prometheus: Add disableRecordingRules datasource config

### LINKED ISSUE #69586: Prometheus: Disable recording and alerting rules api endpoint
The [rules API](https://prometheus.io/docs/prometheus/latest/querying/api/#rules) for some instances can be expensive. 

Allow users to configure and disable recording rules.

### PR DESCRIPTION
**Pull request by shantanualsi**
**What is this feature?**

This PR introduces a config to the Prometheus data source that optionally disables recording rules. It is by default set to `false` so that existing users of the plugin resort to the default behaviour as before. 

**Why do we need this feature?**

The Prometheus recording rules can be expensive for some instances. This PR adds an optional configuration disableRecordingRules to the Prometheus data source.

**Who is this feature for?**

Users of prometheus data source plugin

**Which issue(s) does this PR fix?**:

Fixes #69586

**Special notes for your reviewer:**
I need some help in making the documentation a bit more specific and useful. I think the snippets that I added to the documentation may be somewhat vague. 

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by lwandz13 — APPROVED** [review]
lgtm

**Comment by gtk-grafana** [comment]
@lwandz13 Why are we back porting this feature? I think we typically only backport bugs

**Comment by gtk-grafana** [comment]
@lwandz13 that flow chart says no-backport D1=>D2=>T1. See https://raintank-corp.slack.com/archives/C02GTT9D6CW/p1688052585513679 for more info

**Review by gtk-grafana — APPROVED** [review]
Code looks good, and things are working for me locally. Thanks for taking a look at this!

**Comment by Eve832** [comment]
> @lwandz13 that flow chart says no-backport D1=>D2=>T1. See https://raintank-corp.slack.com/archives/C02GTT9D6CW/p1688052585513679 for more info

Docs is a little different than code in that we support two previous versions of grafana in addition to `latest`. So we backport all changes that apply to those versions.

**Comment by gtk-grafana** [comment]
@Eve832 while this issue has a docs label, it is a new feature, not just docs updates

File: docs/sources/datasources/prometheus/_index.md [hunk 1]
```diff
@@ -141,3 +141,7 @@ This can be toggled on or off in the data source configuration or provisioning f
 Additionally, the amount of overlap between incremental queries can be configured using the `incrementalQueryOverlapWindow` jsonData field, the default value is `10m` (10 minutes).
 
 Increasing the duration of the `incrementalQueryOverlapWindow` will increase the size of every incremental query, but might be helpful for instances that have inconsistent results for recent data.
+
+## Recording Rules (beta)
+
+The Prometheus data source can be configured to disable recording rules under the data souce configuration or provisioning file (under `disableRecordingRules` in jsonData).
```
Comment (gtk-grafana) [inline docs/sources/datasources/prometheus/_index.md:147]:
```suggestion
The Prometheus data source can be configured to disable recording rules under the data source configuration or provisioning file (under `disableRecordingRules` in jsonData).
```

Comment (gtk-grafana) [reply docs/sources/datasources/prometheus/_index.md]: This was causing the drone to fail

**Comment by gtk-grafana** [comment]
Looks like we're hitting an API limit in drone, it resets every hour, I'll rerun the jobs in a bit and get this merged

**Pull request merged**